lib/ansible/plugins/filter/mathstuff.py

```diff
@@ -46,16 +46,16 @@
 
 
 @environmentfilter
-def unique(environment, a, case_sensitive=False, attribute=None):
+def unique(environment, a, case_sensitive=None, attribute=None):
 
     def _do_fail(e):
-        if case_sensitive or attribute:
+        if not case_sensitive and case_sensitive is not None or attribute:
```

Comment:
Can you help me understand why this conditional needs to change?

Reply:
Case_sensitive=None means default true, but none evaluates to false, so also catch the default case.